src/Microsoft.CodeQuality.Analyzers/VisualBasic/QualityGuidelines/BasicRethrowToPreserveStackDetails.vb

```diff
@@ -55,7 +55,10 @@ Namespace Microsoft.CodeQuality.VisualBasic.Analyzers.QualityGuidelines
 
         Private Shared Function IsCaughtLocalThrown(semanticModel As SemanticModel, catchStatement As CatchStatementSyntax, throwExpression As ExpressionSyntax) As Boolean
             Dim local = TryCast(semanticModel.GetSymbolInfo(throwExpression).Symbol, ILocalSymbol)
-            If local Is Nothing OrElse local.Locations.Length = 0 Then
+            Dim catchBlock = TryCast(catchStatement.Parent, CatchBlockSyntax)
+            If local Is Nothing _
+                    OrElse local.Locations.Length = 0 _
+                    OrElse semanticModel.AnalyzeDataFlow(catchBlock?.Statements.First, catchBlock?.Statements.Last).WrittenInside.Contains(local) Then
```

Comment:
API throws an ArgumentNullException for a null value passed in for first or last statement: http://source.roslyn.io/#Microsoft.CodeAnalysis.VisualBasic/Compilation/SemanticModel.vb,c793e537f42cae32.

Reply:
uh...So it is possible that a `catchBlock` is not null but its` statements` are null?

Reply:
The added code has a try cast, so the catchBlock itself can be null. I am also not sure what is the syntax tree structure for an empty block, does it have a null first and/or last statement? It would be good to add a test.

Reply:
Alright, from the caller the `catchStatement` is passed in from `Dim catchStatement = DirectCast(node, CatchBlockSyntax).CatchStatement` such that it would throw NullReference before `IsCaughtLocalThrown` is called. It doesn't make sense to me that the callback is triggered but we have no catchblock. I can add a test for this. But if it is null, shall I just continue to check local?

Reply:
1. Can you change `Dim catchBlock = TryCast(catchStatement.Parent, CatchBlockSyntax)` to a `DirectCast`?
2. Please add a unit test for empty catch block

Reply:
@mavasani  I have updated it to `TryCast` with null check. But for the unit test, this analyzer would not even be invoked for cases like:
```
        Try
            Throw New Exception()
        Catch e As Exception

        End Try
```
Is it still necessary for this unit test?

Reply:
No, it should be fine. Thanks for the contribution.